common/unsafe/src/main/java/org/apache/spark/unsafe/DoubleAccessFunctor.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.unsafe;
+
+import sun.misc.Unsafe;
+
+interface DoubleAccessFunctor {
```

Comment:
Is there any reason to use `interface`? I think that `abstract class` is enough for this.  
I like virtual call rather than interface call. This is because [the code sequnce](https://wiki.openjdk.java.net/display/HotSpot/VirtualCalls) for virtual call is simpler than [that](https://wiki.openjdk.java.net/display/HotSpot/InterfaceCalls) for interface call.